core/modules/wsched/testSchedulers.cc

```diff
@@ -683,28 +694,81 @@ BOOST_AUTO_TEST_CASE(BlendScheduleTest) {
     BOOST_CHECK(blend->getInFlight() == 0);
     BOOST_CHECK(blend->ready() == false);
     LOGS(_log, LOG_LVL_DEBUG, "BlendScheduleTest-2 done");
+
+    LOGS(_log, LOG_LVL_DEBUG, "BlendScheduleTest-3 task/query removal");
+    // Add two queries to scanFast scheduler and then move one query to scanSlow.
+    int startChunk = 70;
+    uint jobs = 11;
+    uint jobsA = jobs;
+    uint jobsB = jobs;
+    std::vector<Task::Ptr> queryATasks;
+    std::vector<Task::Ptr> queryBTasks;
+    lsst::qserv::QueryId qIdA = qIdInc++;
+    lsst::qserv::QueryId qIdB = qIdInc++;
+    {
+        int jobId = 0;
+        int chunkId = startChunk;
+        for (uint j=0; j<jobs; ++j) {
+            taskMsg = newTaskMsgScan(chunkId, lsst::qserv::proto::ScanInfo::Rating::FAST, qIdA, jobId);
+            Task::Ptr mv = makeTask(taskMsg);
+            queryATasks.push_back(mv);
+            queries->addTask(mv);
+            blend->queCmd(mv);
+            taskMsg = newTaskMsgScan(chunkId++, lsst::qserv::proto::ScanInfo::Rating::FAST, qIdB, jobId++);
+            mv = makeTask(taskMsg);
+            queryBTasks.push_back(mv);
+            queries->addTask(mv);
+            blend->queCmd(mv);
+        }
+    }
+    BOOST_CHECK(scanFast->getSize() == jobs*2);
+    BOOST_CHECK(scanSlow->getSize() == 0);
+
+    // This should run 1 job from one of the queries, but there are no guarantees about which one.
+    // This is to test that moveUserQuery behaves appropriately for running Tasks.
+    auto poppedTask = blend->getCmd(false);
+    bool poppedFromA = false;
+    for (auto const& tk : queryATasks) {
+        if (tk == poppedTask) {
+            poppedFromA = true;
+            break;
+        }
+    }
+    if (poppedFromA) --jobsA;
+    else --jobsB;
+
+    blend->moveUserQuery(qIdA, scanFast, scanSlow); // move query qIdA to scanSlow.
+    LOGS(_log, LOG_LVL_DEBUG, "fastSize=" << scanFast->getSize() << " slowSize=" << scanSlow->getSize());
+    BOOST_CHECK(scanFast->getSize() == jobsB);
+    BOOST_CHECK(scanSlow->getSize() == jobsA);
+    // Can't use queryATasks[0] for this as it was popped from the queue before the move.
+    auto taskFromA = queryATasks[1];
+    auto schedForA = std::dynamic_pointer_cast<wsched::ScanScheduler>(taskFromA->getTaskScheduler());
+    LOGS(_log, LOG_LVL_DEBUG, "taskFromA=" << taskFromA->getIdStr() << " sched=" << schedForA->getName());
+    BOOST_CHECK(schedForA == scanSlow);
 }
```

Comment:
This is the longest test case in my memory :scream_cat: Do you think it can be split into >10 smaller cases?